ee/server/service/maintained_apps.go

```diff
@@ -123,12 +123,21 @@ func (svc *Service) AddFleetMaintainedApp(
 		appName = app.Name
 	}
 
+	version := app.Version
+	if version == "latest" { // download URL isn't version-pinned; extract version from installer
+		meta, err := file.ExtractInstallerMetadata(installerTFR)
+		if err != nil {
+			return 0, ctxerr.Wrap(ctx, err, "extracting installer metadata")
+		}
+		version = meta.Version
+	}
```

Comment:
ExtractInstallerMetadata is not rewinding the file properly.
```
	if version == "latest" { // download URL isn't version-pinned; extract version from installer
		meta, err := file.ExtractInstallerMetadata(installerTFR)
		if err != nil {
			return 0, ctxerr.Wrap(ctx, err, "extracting installer metadata")
		}
		version = meta.Version
+		installerTFR.Rewind()
	}
```

Not sure which part of `ExtractInstallerMetadata` is broken, but adding a `installerTFR.Rewind()` fixes our issue.

Reply:
Ah. I thought I had looked at that and then thought it was redundant. My mistake. I can either add this when I'm back at keyboard or feel free to add this without waiting on me.

Reply:
Question I have is.. do we add it here? Or do we want to modify `ExtractInstallerMetadata`? Because if I were using `file.ExtractInstallerMetadata` I would expect it to rewind the file. 

I think in order to get this unblocked we can certainly add it here. But we should at least add a ticket in for fixing `file.ExtractInstallerMetadata`. Feels like a bug to me that it modifies the file.

Reply:
good catch. Unless it's super easy to move the rewind around, happy to just add a rewind in `ee/server/service/maintained_apps.go` and create another ticket to deal with centralizing it. up to you.